dockstore-common/src/main/java/io/dockstore/common/yaml/AbstractYamlService.java

```diff
@@ -69,6 +70,8 @@ public abstract class AbstractYamlService {
 
     private boolean latestTagAsDefault = false;
 
+    private Boolean disableDoiGeneration;
```

Comment:
Not necessary to implement, but another idea is to make this an optional boolean, with `true` and `false` , maybe named something like `generateDoi`, which, when specified, overrides our default choice (as to whether to generate a DOI or not).  Then, you could use it to both opt in and opt out at entry level.

Reply:
As background, this was discussed [here](https://ucsc-gi.slack.com/archives/C05EZH3RVNY/p1727216326027469?thread_ts=1727203321.934499&cid=C05EZH3RVNY).

Our end goal is to default to opt-in (not in this release but the next), and so my thinking was it would be confusing to have an opt-in option that is redundant.

On the other hand, this could allow individuals to opt in individually on their own without Dockstore team having to manipulate the the allow list in Secrets Manager and do a force deploy in the 1.16 release, so maybe we should.